### DISCUSSION TITLE
InitializeComponent(bool) hides inherited member warning.

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#12576] [Q&A] [answered]
I want to make a resizable and draggable control just like controls in Winform Designer:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/949d195c-cd1d-4b09-b717-57c45d2c916b)

Firstly, I created a Usercontrol for base control which achieves drag and resize. As a minimal sample here just use a blank new Usercontrol instead.

Then, I created a child control that inherits it.

Here are the codes of the child control:
```
using Avalonia;
using Avalonia.Controls;
using Avalonia.Markup.Xaml;

namespace AvaloniaApplication3;

public partial class ChildUserControl : BaseControl
{
    public ChildUserControl()
    {
        InitializeComponent();
    }
}
```

```
<Local:ChildUserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
			 xmlns:Local="using:AvaloniaApplication3"
             x:Class="AvaloniaApplication3.ChildUserControl">
  <TextBlock></TextBlock>
</Local:ChildUserControl>
```

Now Visual Studio reports a warnning:

> Severity	Code	Description	Project	File	Line	Suppression State
> Warning	CS0108	'ChildUserControl.InitializeComponent(bool)' hides inherited member 'BaseControl.InitializeComponent(bool)'. Use the new keyword if hiding was intended.	AvaloniaApplication3	C:\Users\Admin\Desktop\AvaloniaApplication3\Avalonia.Generators\Avalonia.Generators.NameGenerator.AvaloniaNameSourceGenerator\AvaloniaApplication3.ChildUserControl.g.cs	18	Active

After I located the issue, I found this code:
```
 public void InitializeComponent(bool loadXaml = true)
 {
     if (loadXaml)
     {
         AvaloniaXamlLoader.Load(this);
     }
 }
```

However, I cannot change this code no matter what I input anything.

In spite the program can run while I want to know what's wrong with my code and how can I clear this warning?

**Comment by IanRawley** [comment, marked as answer]
Have you manually implemented IntializeComponent() in BaseClass?  If so you don't need to, the source generators handle that for you.

**Reply by kyuranger** [reply]
Yes, I have implemented IntializeComponent() in BaseClass. After I deleted it, it works now.

**Reply by sedwick1024** [reply]
I have not, and I still get this warning.  I only call InitializeComponent from constructors.
